pkg/chunked/internal/minimal/compression.go

```diff
@@ -244,7 +244,39 @@ func WriteZstdChunkedManifest(dest io.Writer, outMetadata map[string]string, off
 	return appendZstdSkippableFrame(dest, manifestDataLE)
 }
 
-func ZstdWriterWithLevel(dest io.Writer, level int) (*zstd.Encoder, error) {
+type ZstdWriter interface {
+	io.WriteCloser
+	Reset(dest io.Writer)
+	Flush() error
+}
+
+type CreateZstdWriterFunc func(dest io.Writer) (ZstdWriter, error)
+
+type noCompression struct {
+	dest io.Writer
+}
+
+func (n *noCompression) Write(p []byte) (int, error) {
+	return n.dest.Write(p)
+}
+
+func (n *noCompression) Close() error {
```

Comment:
The way `restartCompression` uses `Close()`+`Reset` is not really fulfilling the `io.WriteCloser` API.

Maybe make this interface a bit higher-level, turning some variant of `restartCompression` into a method on the interface?